src/main/java/seedu/duke/InternshipList.java

```diff
@@ -46,8 +52,12 @@ public boolean isWithinBounds(int index) {
 
     // Method to remove an internship by index (0-based)
     public void removeInternship(int index) {
+        assert isWithinBounds(index) : "Index is out of bounds for removal";
```

Comment:
This assertion seems to throw an error in some use cases. Whenever the user tries to delete an internship whose index is not within the range of the internship list, there seems to be an assertion error thrown. I've pasted the screenshot of one such example input and output below. Please take a look and let me know !

![Screenshot 2024-11-07 122644](https://github.com/user-attachments/assets/cf92c92e-c861-4a09-a4c5-ebd1cb68f3cc)

I think removing this assertion line should fix the error as we cannot assume anything about the range of the internship index at this point.